server/src/main/java/com/cloud/resource/DiscovererBase.java

```diff
@@ -51,6 +54,16 @@
     protected ResourceManager _resourceMgr;
     @Inject
     protected DataCenterDao _dcDao;
+    @Inject
+    private ImageStoreDao imageStoreDao;
+
+    protected void setSecondaryStorageNfsVersionToParams(Long zoneId, Map<String, Object> params) {
+        ImageStoreVO imageStoreInZone = imageStoreDao.findOneByZoneAndProtocol(zoneId, "nfs");
```

Comment:
@nvazquez Are multiple image stores with different nfs versions supported in a zone?

Reply:
Potentially yes - suggesting we should pick the lower value set in this case? cc @DaanHoogland

Reply:
@nvazquez @sureshanaparti I do not see a problem here. Unless I am looking at the code in isolation too much and should be looking at a bigger picture, what happens here:
```
a store is picked
if a store is found
- the nfs version for that store is retrieved and used
if not
- the default nfs version is used
```
what could go wrong here is only that the operator has not set the right nfs versions to the system/stores.